ssl/quic/quic_wire.c

```diff
@@ -497,6 +497,18 @@ int ossl_quic_wire_peek_frame_ack_num_ranges(const PACKET *orig_pkt,
             || !PACKET_get_quic_vlint(&pkt, &ack_range_count))
         return 0;
 
+    /*
+     * Ensure the specified number of ack ranges listed in the ACK frame header
+     * actually are available in the frame data. This naturally bounds the
+     * number of ACK ranges which can be requested by the MDPL, and therefore by
+     * the MTU. This ensures we do not allocate memory for an excessive number
+     * of ACK ranges.
+     */
+    for (i = 0; i < ack_range_count; ++i)
+        if (!PACKET_skip_quic_vlint(&pkt)
+            || !PACKET_skip_quic_vlint(&pkt))
+            return 0;
```

Comment:
Not knowing much about QUIC, can other data be behind this? Should we check that after this we actually have 0 bytes left in the pkt?

Reply:
Yes, there are multiple frames in a datagram. So totally natural for there to be data left over.